### ISSUE TITLE
support FakeTensor input for torch.compile

### ISSUE THREAD

**Issue by weifengpy** [pytorch/pytorch#136586]
### 🚀 The feature, motivation and pitch

In AutoFSDP, I want to estimate flops and runtimes without running model on real GPUs. I was trying to use FakeTensor input for torch.compile. I got following error from `backend=inductor` (`backend=eager` works)

Wonder the workload to make it work? There are maybe 2 fake modes, one from my user code, and one from PT2 stack

```
  File "/data/users/weif/pytorch/torch/_dynamo/symbolic_convert.py", line 983, in run
    while self.step():
  File "/data/users/weif/pytorch/torch/_dynamo/symbolic_convert.py", line 895, in step
    self.dispatch_table[inst.opcode](self, inst)
  File "/data/users/weif/pytorch/torch/_dynamo/symbolic_convert.py", line 2987, in RETURN_VALUE
    self._return(inst)
  File "/data/users/weif/pytorch/torch/_dynamo/symbolic_convert.py", line 2972, in _return
    self.output.compile_subgraph(
  File "/data/users/weif/pytorch/torch/_dynamo/output_graph.py", line 1117, in compile_subgraph
    self.compile_and_call_fx_graph(tx, list(reversed(stack_values)), root)
  File "/data/users/weif/pytorch/torch/_dynamo/output_graph.py", line 1360, in compile_and_call_fx_graph
    backend_fake_mode = torch._subclasses.FakeTensorMode(
  File "/data/users/weif/pytorch/torch/_subclasses/fake_tensor.py", line 1174, in __init__
    self._stack_trace = traceback.extract_stack()

fake mode from fake tensor input 0 allocated at:
  File "/data/users/weif/pytorch/test_fake_tensor_compile.py", line 5, in <module>
    with FakeTensorMode():
  File "/data/users/weif/pytorch/torch/_subclasses/fake_tensor.py", line 1174, in __init__
    self._stack_trace = traceback.extract_stack()
```

import torch
import torch.nn as nn
from torch._subclasses.fake_tensor import FakeTensorMode

with FakeTensorMode():
    model = nn.Linear(4, 4, device="cuda")
    inp = torch.rand(4, 4, device="cuda")
    loss = torch.compile(model)(inp).sum()
    loss.backward()
```

### Alternatives

_No response_

### Additional context

_No response_

**Comment by ezyang** [comment]
@eellison didn't you already file an issue for this

**Comment by weifengpy** [comment]
> Yes: #136446. but this is an additional request for flops

it's nice to see the same issue. I was mainly asking for composability with FakeTensorMode. I mentioned flops to cover context on why I am asking. It does not have to be solved together

**Comment by exclamaforte** [comment]
@weifengpy is running with this backend enough for your purposes?
https://github.com/pytorch/pytorch/pull/143250
